File: WebTs/src/services/categoryService.ts
// services/categoryService.ts
import axios from "axios";
import { Category } from "../types/Category";

const BASE_URL = "http://localhost:9090/category/all";

export const getAllCategories = async (): Promise<Category[]> => {
  const response = await axios.get<{ categories: Category[] }>(BASE_URL);
  return response.data.categories;
};

File: MyBackEnd/src/main/java/com/example/mybackend/Services/CategoryService.java
package com.example.mybackend.Services;

import com.example.mybackend.Models.Category;
import com.example.mybackend.Repositories.CategoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Service
public class CategoryService {
    @Autowired
    CategoryRepository categoryRepository;

    public Map<String, Object> getAllCategories() {
        Map<String, Object> response = new HashMap<>();
        List<Category> categories = categoryRepository.findAll();
        if ((categories == null) || (categories.size() == 0)) {
            response.put("message", "No categories found");
            return response;
        }
        response.put("categories", categoryRepository.findAll());
        return response;
    }
}
